ONSdigital/response-operations-ui
# Removed reliance on node-sass and swapped to dart sass

# Motivation and Context
Pipeline had broken due to a reliance by an NPM package (`gulp-sass`) on a no-longer-existent binary hosted on Github.

Have moved to a non-compiled version, because there is a current security advisory to do with the `tar` package, which is pinned by `node-gyp`.  To mitigate this, I have taken out compiled steps from the NPM packages, so that `node-gyp` will not be used.

Once `node-gyp` address their reliance on `tar` @2.2.8 (current version is 4.4.8, and is the first that removes the vulnerability), we can consider using compiled bindings.

# What has changed
* `node-sass` module removed
* `sass` module (dart-sass) added
* `sass` specified as `gulp-sass` compiler
* `fibers` module used to speed up compilation loss.

# How to test?
Ensure that you can still run gulp tasks, particularly `scss`, and `build`.
Ensure that you can still run `make build` with success.

# Links
* https://www.npmjs.com/advisories/803
* https://trello.com/c/tLmUCrGE
